### DISCUSSION TITLE
Begin recording (from analog) when someone speaks

### DISCUSSION THREAD

**Discussion by imargonis** [pschatzmann/arduino-audio-tools#382] [Q&A] [answered]
Is there a way to make recording dependent on someone talking (I guess this means having some threshold or something)?

I am using an analog microphone (DFRobot SEN0487 https://wiki.dfrobot.com/Fermion_MEMS_Microphone_Sensor_SKU_SEN0487) with an ESP32 (switched from an ESP8266 that is not supported) to try to record audio and save it to an SD card and also send through wifi (admittedly I haven't managed to make this work yet).

Is there a way to trigger the recording only when someone speaks. Most examples I've seen just record for a predefined time and that's it (or rely on button push).

**Comment by pschatzmann** [comment, marked as answer]
If you check the output classes there is the [VolumePrint](https://pschatzmann.github.io/arduino-audio-tools/html/classaudio__tools_1_1_volume_print.html) class to determine the volume...
